File: internal/content/testdata/bench/go/nc-coin-change-2-not.go
func max_product(nums []int) int {
	maxProd, minProd, res := nums[0], nums[0], nums[0]
	for _, n := range nums[1:] {
		a, b, c := n, maxProd*n, minProd*n
		maxProd = max(a, max(b, c))
		minProd = min(a, min(b, c))
		res = max(res, maxProd)
	}
	return res
}

File: internal/content/testdata/bench/typescript/nc-coin-change-2-not.ts
function max_product(nums: number[]): number {
    let max_prod = nums[0];
    let min_prod = nums[0];
    let res = nums[0];
    for (let i = 1; i < nums.length; i++) {
        const n = nums[i];
        const candidates = [n, max_prod * n, min_prod * n];
        max_prod = Math.max(...candidates);
        min_prod = Math.min(...candidates);
        res = Math.max(res, max_prod);
    }
    return res;
}
